setup.py

```diff
@@ -1,10 +1,25 @@
 import pathlib
+import os
 from setuptools import setup
 from setuptools import find_packages
 
 with open("README.md", encoding="utf-8") as readme_file:
     long_description = readme_file.read()
 
+# The directory containing this file
+my_current_dir = pathlib.Path(__file__).parent
+
+# Pull requirements from the text file
+requirement_path = os.path.join(my_current_dir, "requirements.txt")
+
+install_requires = []
+if os.path.isfile(requirement_path):
```

Comment:
if this fails (i.e. `requirement_path` is not a file) , then it looks like nothing installs. We don't want that right? It will push to pypi and then not check requirements upon install?

Maybe instead add an assertion that it is a file?

Reply:
(i've wanted to do this for a while btw - reduce redundancy between this script and requirements, so, thank you!)

Reply:
If line 16 doesn't catch it as a file, then the install_requires is empty by line 15. We could assert, but we'll check it clearnly on line 16 anyway.

Reply:
but we don't want `install_requires` to be empty, right?

My concern is that this when this fails (i.e. an empty `install_requires`) it will fail silently. Maybe just removing line 16 is sufficient, since then if it fails, it will fail loudly